Assets/Scripts/SceneLoader.cs

```diff
@@ -1,113 +1,126 @@
+            playerMinSaves = PlayerPrefs.GetString("playerMinSaves");
         }
         else
         {
-            starprogress = "0;-1;-1;-1;-1;-1;-1;-1;-1;-1";
-            PlayerPrefs.SetString("starprogress", starprogress);
+            PlayerPrefs.SetString("playerMinSaves", playerMinSaves);
         }
+    }
 
-        string[] starSplit = starprogress.Split(';');
+    // Populate menu stars according to playerMinSaves and savesRequired
+    private void PopulateStars()
+    {
+        int[] minSaves = Array.ConvertAll(playerMinSaves.Split(';'), s=> int.Parse(s));
         int currentLevel = 1;
-        foreach (string star in starSplit)
+        foreach (int saves in minSaves)
         {
-            GameObject level = Instantiate(LevelSelector, LevelContent.transform.position, Quaternion.identity);
+            List<int> savesNeeded = savesRequired[currentLevel-1];
 
+            GameObject level = Instantiate(LevelSelector, LevelContent.transform.position, Quaternion.identity);
             level.transform.parent = LevelContent.transform;
             level.transform.localScale = new Vector3(1, 1, 1);
+            level.name = currentLevel.ToString() + "-level" + currentLevel.ToString();
 
-            if (star == "-1")
+            if (saves == -1) // locked level
             {
-                level.name = currentLevel.ToString() + "-level" + currentLevel.ToString();
                 level.transform.GetChild(0).gameObject.SetActive(false);
                 level.transform.GetChild(1).gameObject.SetActive(true);
                 level.transform.GetChild(2).gameObject.SetActive(false);
                 level.transform.GetChild(3).gameObject.SetActive(false);
                 level.transform.GetChild(4).gameObject.SetActive(false);
+                continue;
             }
-            else if (star == "0")
-            {
-                level.name = currentLevel.ToString() + "-level" + currentLevel.ToString();
-                level.GetComponent<UnityEngine.UI.Button>().onClick.AddListener(() =>
-                {
-                    LoadScene(level.name);
-                });
 
+            level.GetComponent<UnityEngine.UI.Button>().onClick.AddListener(() =>
+            {
+                LoadScene(level.name);
+            });
+            if (saves == 0) // not attempted
+            {
                 level.transform.GetChild(0).gameObject.SetActive(true);
                 level.transform.GetChild(0).gameObject.GetComponent<TextMeshProUGUI>().text = currentLevel.ToString();
                 level.transform.GetChild(1).gameObject.SetActive(false);
                 level.transform.GetChild(2).gameObject.SetActive(false);
                 level.transform.GetChild(3).gameObject.SetActive(false);
                 level.transform.GetChild(4).gameObject.SetActive(false);
             }
-            else if (star == "1")
+            else if (saves <= savesNeeded[2]) // 3 stars
```

Comment:
Put all the `if` logic for checking number of stars into a function that returns the level STATE_NUMBER.